### DISCUSSION TITLE
How to get best timing consistency for midi player?

### DISCUSSION THREAD

**Discussion by SubhadeepJasu** [FluidSynth/fluidsynth#1009] [Q&A] [answered]
What settings should I use to ensure that the MIDI Player has the best timing / tempo consistency even at the cost of audio quality?

[ I am using pulseaudio and alsa as audio drivers. This problem doesn't exist with jack.]

**Comment by derselbst** [comment, marked as answer]
It is not clear to me, what "problem" you are referring to. And it's not clear either, what makes you think that this "problem" doesn't exist with jack. Hence my only advice is to use `-o player.timing-source=sample` (which is the default value). Fluidsynth will take care of accurate MIDI event dispatching.

**Reply by SubhadeepJasu** [reply]
I see. well sample it is.
